app/assets/javascripts/calls.coffee

```diff
@@ -1,3 +1,40 @@
 # Place all the behaviors and hooks related to the matching controller here.
 # All this logic will automatically be available in application.js.
 # You can use CoffeeScript in this file: http://coffeescript.org/
+
+jQuery -> 
+
+    $('#call_list td').each (index, element) ->
+      $(element).css('width', $(element).width())
+
+    $('#call_list').sortable(
+      placeholder: "ui-state-highlight"
+      axis: 'y'
+      items: '.pregnancy-data'
+      handle: '.handle'
+      cursor: 'move'
+      sort: (e, ui) ->
+        ui.item.addClass('active-item-shadow')
+      stop: (e, ui) ->
+        ui.item.removeClass('active-item-shadow')
+        # highlight the row on drop to indicate an update
+        ui.item.children('td').effect('highlight', {}, 500)
+      update: (e, ui) ->
+        item_id = ui.item[0].id
+        parent = $('#'+item_id).parent()
+        children = parent.children()
+        order = []
+        for child, value of children
+          if value.id? then order.push(value.id)
+        console.log(order)
+
+        $.ajax(
+          type: 'POST'
+          url: '/something/reorder_pregnancies'
+          dataType: 'json'
+          data: { order }
+        )
```

Comment:
This looks slamming, thanks bud!

I was at work a little later than I anticipated today. I'll start plugging
our respective prs into each other tomorrow.

On Tue, Aug 2, 2016, 5:07 PM Mike Chelen <notifications@github.com> wrote:

> In app/assets/javascripts/calls.coffee
> <https://github.com/colinxfleming/dcaf_case_management/pull/559#discussion_r73237866>
> :
>
> > +        ui.item.children('td').effect('highlight', {}, 500)
> > +      update: (e, ui) ->
> > +        item_id = ui.item[0].id
> > +        parent = $('#'+item_id).parent()
> > +        children = parent.children()
> > +        order = []
> > +        for child, value of children
> > +          if value.id? then order.push(value.id)
> > +        console.log(order)
> > +
> > +        $.ajax(
> > +          type: 'POST'
> > +          url: '/something/reorder_
>
> pregnancies'
> > +          dataType: 'json'
> > +          data: { order }
> > +        )
>
> This is the last remaining part, setting the proper URL here and the post
> data structure.
>
> —
>
>
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/colinxfleming/dcaf_case_management/pull/559/files/c0df880baa0a94b2092b2da6b772bf75c475d7f4#r73237866>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADsA9F0ruT9L8zBuYfJ6tsauDRjKPFcmks5qb7GdgaJpZM4JbC-m>
> .
>